pkg/controller/linstorcontroller/linstorcontroller_controller.go

```diff
@@ -1056,12 +1056,24 @@ func expectedEndpoint(controllerResource *piraeusv1.LinstorController) string {
 
 func getObjectMeta(controllerResource *piraeusv1.LinstorController, nameFmt string) metav1.ObjectMeta {
 	return metav1.ObjectMeta{
-		Name:      fmt.Sprintf(nameFmt, controllerResource.Name),
-		Namespace: controllerResource.Namespace,
-		Labels: map[string]string{
-			"app.kubernetes.io/name":       kubeSpec.ControllerRole,
-			"app.kubernetes.io/instance":   controllerResource.Name,
-			"app.kubernetes.io/managed-by": kubeSpec.Name,
-		},
+		Name:        fmt.Sprintf(nameFmt, controllerResource.Name),
+		Namespace:   controllerResource.Namespace,
+		Labels:      getControllerLabels(controllerResource),
+		Annotations: controllerResource.ObjectMeta.Annotations,
+	}
+}
+
+func getControllerLabels(controllerResource *piraeusv1.LinstorController) map[string]string {
+	return mergeStringMap(map[string]string{
+		"app.kubernetes.io/name":       kubeSpec.ControllerRole,
+		"app.kubernetes.io/instance":   controllerResource.Name,
+		"app.kubernetes.io/managed-by": kubeSpec.Name,
+	}, controllerResource.ObjectMeta.Labels)
```

Comment:
We have to be careful not to include these "user" labels in the `Selector:` for deployments. Otherwise we could get into a situation where a user modified a label, but the operator can't proceed, because updating the selector value isn't possible.